File: TransCoder_tokenized_test_set_functions/python/NUMBER_DIGITS_REMOVED_MAKE_NUMBER_DIVISIBLE_3.py
def divisible ( num ) :
    n = len ( num );
    
    sum = 0;
    
    for i in range ( n ) :
        sum += int ( num [ i ] );
        
    if ( sum % 3 == 0 ) :
        return 0;
        
    if ( n == 1 ) :
        return - 1;
        
    for i in range ( n ) :
        if ( sum % 3 == int ( num [ i ] ) % 3 ) :
            return 1;
            
    if ( n == 2 ) :
        return - 1;
        
    return 2;

File: results/py/NUMBER_DIGITS_REMOVED_MAKE_NUMBER_DIVISIBLE_3.cpp
int f_filled ( string num ) {
  int n = num . length ( ) ;
  int sum = 0 ;
  ;
  for ( int i = 0 ;
  i < n ;
  i ++ ) sum += atoi ( num . c_str ( ) ) ;
  if ( ( sum % 3 == 0 ) && ( sum % 2 == 0 ) ) return 0 ;
  if ( ( n == 1 ) && ( sum % 3 == ( int ) num . length ( ) % 3 ) ) return 1 ;
  for ( int i = 0 ;
  i < n ;
  i ++ ) if ( ( sum % 3 == ( int ) num . length ( ) % 3 ) ) return - 1 ;
  if ( ( n == 2 ) && ( sum % 2 == ( int ) num . length ( ) % 2 ) ) return - 1 ;
  return 2 ;
}
